packages/http-client-csharp/emitter/src/type/input-examples.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { InputParameter } from "./input-parameter.js";
+import {
+  InputArrayType,
+  InputDictionaryType,
+  InputLiteralType,
+  InputModelType,
+  InputNullableType,
+  InputPrimitiveType,
+  InputType,
+  InputUnionType,
+} from "./input-type.js";
+import { OperationResponse } from "./operation-response.js";
+
+interface InputExampleBase {
+  kind: string;
+  name: string;
+  description: string;
+  filePath: string;
+  rawExample: any;
+}
+
+export interface InputHttpOperationExample extends InputExampleBase {
+  kind: "http";
+  parameters: InputParameterExample[];
+  responses: Map<number, OperationResponseExample>;
+}
+
+export interface InputParameterExample {
```

Comment:
`InputParameterExample` and `InputOperationResponseExample` are not `InputExampleBase`. But they are used by `InputHttpOperationExample` which is `InputExampleBase`, and they refer to `InputTypeExample` which is also `InputExampleBase`.
That's because `InputParameterExample` and `InputOperationResponseExample` cannot be referenced as the root of examples, but all others can?

Reply:
TCGC defines those types exactly in this way.
Also this is typescript, please do not take the inheritance hierarchy as a big deal, it makes no sense in ts context because of duck types.
`InputTypeExample` should not have the things in the `InputExampleBase`, `name` `description` all makes no sense.

Their types might not be properly named.
Those types can be categorize into two different types:
1. the example
2. the value
They did not name them properly to distinguish these two categories therefore it confuses you.

The `InputHttpOperationExample` is the example itself, every other types are "example values"

Reply:
I think so. Those two types probably should be named as `InputParameterExampleInstance` and `InputOperationResposeExampleInstance`.

Reply:
well I chnaged those two to `XXXValue`, I think this should also align with the `@example` design in typespec as well, the parameter and response are just complex object values here.